### ISSUE TITLE
Authenticator (React) remounts the app after refreshing the access token

### ISSUE THREAD

**Issue by neil-a-thomson** [aws-amplify/amplify-ui#2191]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Webpack

### Please describe your bug.

The components provided to `withAuthenticator` are remounted when the access token is refreshed from the userpool.

### What's the expected behaviour?

I expect the app to be rerendered when a new token is fetched, I do not expect the whole app to remount.

### Code Snippet

```typescript
import React, { useCallback, useEffect } from 'react';
import { withAuthenticator } from '@aws-amplify/ui-react';
import { Auth } from 'aws-amplify';

function App({ signOut, user }) {
  useEffect(() => {
    // this gets called every time the token expires because the app remounts
    // I'm pretty sure this isn't the intended behaviour
    console.log('App mounting');
  }, []);

  const apiCall = useCallback(async () => {
    console.log('Making API call');
    const session = await Auth.currentSession(); // refreshes session if expired
    // don't need to do anything more, the line above is enough to cause a remount
  }, []);

  return (
    <div>
      <div>App</div>
      <div>{user.username}</div>
      <div>
        <button onClick={() => apiCall()}>Click me</button>
      </div>
    </div>
  );
}

export default withAuthenticator(App);
```

**Comment by HudsonAkridge** [comment]
We're also running into this exact same issue. When performing a call to:
```JavaScript
let cognitoSession = await Auth.currentSession();
let idToken = cognitoSession.idToken.jwtToken;
```

in the background to capture our user's JWT's so we can issue API calls directly to our app, and the JWT needs to be refreshed, the entire app under the
```JavaScript
        <Authenticator
            loginMechanisms={["username"]}
            hideSignUp={true}
        >
            <Router>
                <Etc/>
            </Router>
        </Authenticator>
```
wrapper is forced to re-mount, losing state across the entire SPA.

Extremely annoying for our users. Taking away the calls to get the `jwtToken` does not appear to prevent the remount due to a refresh, it just changes where the refresh causing remount happens.

If we open a new tab/window and access the app there, the refresh fires inside that new tab/window, and then we close that tab/window to return to our original SPA session, we can continue using the app in the original state without having the components remounted. This is obviously sub-optimal and not possible or expected behavior to train our users on.

We are using the cookie based storage set up as such:

```JavaScript
amplifyAwsConfig.cookieStorage = {
    domain: ApiConfig.hostname,
    path: "/",
    expires: 365,
    sameSite: "lax",
    secure: ApiConfig.hostname !== "localhost",
};
```

The JWT Refresh causing `Authenticator` remount issue happens whether our cookie storage is configured like the above or not.

The only workaround we can come up with is to track the expiration ticks for the JWT when we first log in, and then to have a background worker try and open up a new tab/window to refresh the auth token when it expires ahead of our main SPA page the user is on. We'd like to avoid needing to do this, as it adds a decent bit of unnecessary complication to things, especially if they've got multiple tabs/windows open already working in different areas of our app.

The ideal fix would be if the `amplify-ui` library could be updated to correct this behavior for us. Happy to provide any additional details to assist with closing out this issue. Thank you!

**Comment by reesscot** [comment]
Thank you for reporting this issue. We are looking into it.

**Comment by mroest** [comment]
Same issue for us :( Children of the Authenticator are remounted for every render. Which is a bit issue for us since that breaks our SPA. We have a graphql backend and need to send the jwt token to the API. Also in the process validate the session and therefore doing a Auth.currentSession() for an API request.

Please have look at this. We've tried different versions 2.x and 3.x. The issue exists in all versions we've tried.

**Comment by mroest** [comment]
Confirmed 2.15.8 works for us!

**Comment by calebpollman** [comment]
@neil-a-thomson There is a potential fix in this tagged release:`@aws-amplify/ui-react@authenticator-remount-2191`. Could you install and check if it resolves your issue?

**Comment by reesscot** [comment]
@mroest and @aterlamia Would you please test the potential fix @calebpollman put together?

```
npm install @aws-amplify/ui-react@authenticator-remount-2191
```

**Comment by aterlamia** [comment]
I can confirm that using remount-2181 works for me

**Comment by reesscot** [comment]
@aterlamia Thanks for testing, we'll let you know when the fix is released to `latest`.

**Comment by reesscot** [comment]
The fix has now been released in version`3.0.4`

```
npm install @aws-amplify/ui-react@3.0.4
```
@HudsonAkridge @aterlamia @mroest @neil-a-thomson

**Comment by HudsonAkridge** [comment]
Thank you @reesscot !

We've got it running on our site now, will let you know if it solves the issue, but if it solved it for @aterlamia , then it should solve it for us as well.
